datacube/drivers/postgis/_core.py

```diff
@@ -29,7 +33,41 @@
 )
 from datacube.migration import ODC2DeprecationWarning
 
-USER_ROLES = ("odc_user", "odc_manage", "odc_admin")
+
+class UserRole(Enum):
+    USER = "odc_user"
+    MANAGE = "odc_manage"
+    ADMIN = "odc_admin"
+
+    @classmethod
+    def to_pg_role(cls, role_str: Literal["user", "manage", "admin"]) -> "UserRole":
+        return cls("odc_" + role_str.lower())
+
+    def simple_str(self) -> str:
+        return self.value.split("_")[1]
```

Comment:
Shouldn't this be `"_".join(self.value.split("_")[1:])` so roles can have `_` in their names?

Reply:
We only have the 3 (or 4) role names (user, manage, admin, and ingest in the legacy driver) and it seems unlikely we will need more - certainly not enough to require `_` in their names.

No real harm I guess, but can be done more concisely with the maxsplit arg to `split`.